sdcm/cluster.py

```diff
@@ -3586,19 +3586,29 @@ def cql_connection_patient_exclusive(self, node, keyspace=None,
 
     def get_non_system_ks_cf_list(self, db_node,  # pylint: disable=too-many-arguments
                                   filter_out_table_with_counter=False, filter_out_mv=False, filter_empty_tables=True,
-                                  filter_by_keyspace: list = None) -> List[str]:
+                                  filter_by_keyspace: list = None, filter_out_by_replication_factor: int | None = None) -> List[str]:
         return self.get_any_ks_cf_list(db_node, filter_out_table_with_counter=filter_out_table_with_counter,
                                        filter_out_mv=filter_out_mv, filter_empty_tables=filter_empty_tables,
-                                       filter_out_system=True, filter_out_cdc_log_tables=True, filter_by_keyspace=filter_by_keyspace)
+                                       filter_out_system=True, filter_out_cdc_log_tables=True, filter_by_keyspace=filter_by_keyspace,
+                                       filter_out_by_replication_factor=filter_out_by_replication_factor)
 
-    def get_any_ks_cf_list(self, db_node,  # pylint: disable=too-many-arguments
+    def get_any_ks_cf_list(self, db_node,  # pylint: disable=too-many-arguments,too-many-statements
                            filter_out_table_with_counter=False, filter_out_mv=False, filter_empty_tables=True,
                            filter_out_system=False, filter_out_cdc_log_tables=False,
-                           filter_by_keyspace: list = None) -> List[str]:
+                           filter_by_keyspace: list = None, filter_out_by_replication_factor: int | None = None) -> List[str]:
         regular_column_names = ["keyspace_name", "table_name"]
         materialized_view_column_names = ["keyspace_name", "view_name"]
         regular_table_names, materialized_view_table_names = set(), set()
         where_clause = ""
+        # This filter is usefull when the some
+        if filter_out_by_replication_factor:
+            with self.cql_connection_patient(db_node, connect_timeout=600) as session:
+                result = list(session.execute("select keyspace_name, replication from system_schema.keyspaces"))
+                filter_by_keyspace = [row.keyspace_name for row in result
```

Comment:
logic here is a bit weird:
for example `filter_out_by_replication_factor=2`, it would pick KS with rf=1 and KS with rf=3

I would think it's maybe bigger than 1, or bigger then 2

also I would put the logic of retrieving the RF, on it's own utility function, and run this at the end, after all the other filters

Reply:
overwriting `filter_by_keyspace` meaning you can use those options at the same time..

Reply:
I cannot think about another example when this filter may be needed. 
If it may be case when we want to filter out RF = 2, but RF 1 or 3 is OK. What is the reason to filter out RF 1 together with RF 2?

Reply:
we don't have such a case now, but if we introduce a feature into our testing, we should strive for it to be concise and clear logic.

we don't have cases that used RF=2 now, but we didn't had one that use RF=1 till this one.
one possible use case, if for when we need to change RF, before decommission. we should be able to look for the keyspaces which have `RF=number of nodes` so we can ALTER it. (see https://github.com/scylladb/scylla-cluster-tests/pull/7457 as an example)

Reply:
Here it is another case. 
It is opposite: we need to find any keyspace that RF is not equal (actually, pretty clear logic). So logic "select that more then" or "equal" is not suitable for this case.
If you do not like the solution from this commit, I can create additional function that go over received function and filter them again (that will need additional query of DB)